### ISSUE TITLE
Dual stack bind address

### ISSUE THREAD

**Issue by kyrofa** [k3s-io/k3s#8782]
By default, `k3s server` listens on port 6443 on all interfaces. I have a cluster where the nodes:

1. Are dual stack, i.e. they have both ipv4 and ipv6
2. Have two NICs, one public, one private

Except for my ingress controller, I want everything listening on the private interface, so I ended up using a number of CLI options to accomplish that. The one I'm talking about here is `--bind-address`. By default it's 0.0.0.0, so it listens to everything. In order to NOT listen to everything, I used `--bind-address`, but it only accepts a single value, so I had to choose ipv4 or ipv6. Ideally I would be able to have it listen on both IP addresses for the private NIC, but that doesn't appear possible.

`--node-ip` and `--node-external-ip` both support comma-separated lists, but trying to do the same for `--bind-address` just results in "too many colons". Is this something that k3s can support? Or am I misunderstanding, and there's already another way to do this?

I'm using k3s v1.27.6+k3s1, by the way.

**Comment by brandond** [comment]
No, there is not currently any way to specify dual-stack bind addresses. We use https://github.com/rancher/dynamiclistener and it currently supports binding to a single address; we would need to update that library to take a list of addresses first, and then make use of that here.

**Comment by kyrofa** [comment]
Thanks @brandond. We figured we'd run into some rough edges using dual stack!

**Comment by kevinvalk** [comment]
**TL;DR; Make sure to provide a single address to `bind-address` and in the case of dual-stack ensure it is IPv4**

For anyone else facing hours debugging this. In the case of dual stack it seems you **HAVE TO** to use an IPv4 bind address. What's more is that you can do the following in `config.yaml` without any apparent issues:
```yaml
bind-address:
- 10.0.1.5
- fdda:c2b1:1fd2:4::1
```

However, in this case only the **last** IP provided in the list is used for the actual bind address.

P.S. I believe this is a known issue and documented at https://docs.k3s.io/networking/basic-network-options#dual-stack-ipv4--ipv6-networking under the bottom "known issue". Sadly, it simply did not click for me that this was that...

> When defining cluster-cidr and service-cidr with IPv6 as the primary family, the node-ip of all cluster members should be explicitly set, placing node's desired IPv6 address as the first address. By default, the kubelet always uses IPv4 as the primary address family.

**Comment by kyrofa** [comment]
Hmm, I'm using an ipv6 bind address without issue.

**Comment by kevinvalk** [comment]
> Hmm, I'm using an ipv6 bind address without issue.

I think that is intended if you provide node-ip, CIDRs ranges, etc with IPv6 family first and second entry is IPv4. But please confirm or deny as I do think that is valuable information!
